contracts/modules/TransferManager/BlacklistTransferManager.sol

```diff
@@ -0,0 +1,244 @@
+    * @param _endDate end date of the blacklist type
+    * @param _name name of the blacklist type
+    * @param _repeatPeriodInDays repeat period of the blacklist type
+    */
+    function modifyBlacklistType(uint256 _startDate, uint256 _endDate, bytes32 _name, uint256 _repeatPeriodInDays) public withPerm(ADMIN){
+        require(blacklists[_name].endDate != 0, "Blacklist type doesn't exist"); 
+        _validParams(_startDate, _endDate, _name, _repeatPeriodInDays);
+        blacklists[_name] = BlacklistsDetails(_startDate, _endDate, _repeatPeriodInDays);
+        emit ModifyBlacklistType(_startDate, _endDate, _name, _repeatPeriodInDays);
+    }
+
+    /**
+    * @notice Used to delete the blacklist type
+    * @param _name name of the blacklist type
+    */
+    function deleteBlacklistType(bytes32 _name) public withPerm(ADMIN){
+        require(blacklists[_name].endDate != 0, "Blacklist type doesn’t exist");
+        require(blacklistToAddress[_name].length == 0, "Investors are associated with the blacklist");
+        delete(blacklists[_name]);
+        emit DeleteBlacklistType(_name);
+    }
+
+    /**
+    * @notice Used to assign the blacklist type to the investor
+    * @param _investor address of the investor
+    * @param _blacklistName name of the blacklist
+    */
+    function addInvestorToBlacklist(address _investor, bytes32 _blacklistName) public withPerm(ADMIN){
+        require(blacklists[_blacklistName].endDate != 0, "Blacklist type doesn't exist");
+        require(_investor != address(0), "Invalid investor address");
+        investorToBlacklist[_investor].push(_blacklistName);
+        blacklistToAddress[_blacklistName].push(_investor);
+        emit AddInvestorToBlacklist(_investor, _blacklistName);
+    }
+
+     /**
+    * @notice Used to delete the investor from the all associated blacklists
+    * @param _investor address of the investor
+    */
+    function deleteInvestorFromAllBlacklist(address _investor) public withPerm(ADMIN){
+        require(_investor != address(0), "Invalid investor address");
+        require(investorToBlacklist[_investor].length != 0, "Investor is not associated to any blacklist type");
+        for(uint256 i = 0; i < investorToBlacklist[_investor].length; i++){
+            deleteInvestorFromBlacklist(_investor, investorToBlacklist[_investor][i]);
+        }
+    }
+
+     /**
+    * @notice Used to delete the investor from the blacklist
+    * @param _investor address of the investor
+    * @param _blacklistName name of the blacklist
+    */
+    function deleteInvestorFromBlacklist(address _investor,bytes32 _blacklistName) public withPerm(ADMIN){
+        require(_investor != address(0), "Invalid investor address");
+        require(_blacklistName != bytes32(0),"Invalid blacklist name");
+        require(investorToBlacklist[_investor].length != 0, "Investor is not associated to any blacklist type");
+        for(uint256 j = 0; j < investorToBlacklist[_investor].length; j++){
```

Comment:
Would prefer to avoid unbounded loops if possible - could maintain an index for where the investor / blacklist is in each of the arrays to do this.